SPEC.rdoc

```diff
@@ -260,31 +279,63 @@ The values of the header must be Strings,
 consisting of lines (for multiple header values, e.g. multiple
 <tt>Set-Cookie</tt> values) separated by "\\n".
 The lines must not contain characters below 037.
+
 === The Content-Type
+
 There must not be a <tt>Content-Type</tt>, when the +Status+ is 1xx,
 204 or 304.
+
 === The Content-Length
+
 There must not be a <tt>Content-Length</tt> header when the
 +Status+ is 1xx, 204 or 304.
+
 === The Body
-The Body must respond to +each+
-and must only yield String values.
 
-The Body itself should not be an instance of String, as this will
-break in Ruby 1.9.
+The Body is typically an Array of Strings, the application instance
+itself, or a File-like object.
+
+The Body must respond to +each+ or +call+. It may optionally respond to
++to_path+.
+
+A Body that responds to +each+ is considered to be a chunked body. If
+it also responds to +to_path+ is considered to be the same as that file
+the local storage.
+
+A Body that responds to +call+ is considered to be a streaming body.
 
-If the Body responds to +close+, it will be called after iteration. If
-the body is replaced by a middleware after action, the original body
-must be closed first, if it responds to close.
+If the Body responds to +close+, it will be called after iteration.
+If the body is replaced by a middleware after action, the original
+body must be closed first, if it responds to close.
+
+==== Chunked Body
+
+The Chunked Body must respond to +each+.
+It must only be called once.
+and must only yield String values.
 
 If the Body responds to +to_path+, it must return a String
 identifying the location of a file whose contents are identical
 to that produced by calling +each+; this may be used by the
 server as an alternative, possibly more efficient way to
 transport the response.
 
-The Body commonly is an Array of Strings, the application
-instance itself, or a File-like object.
+
+==== Streaming Body
+
+The Streaming Body must respond to +call+.
+It must only be called once.
+It takes a +stream+ argument.
+
+The +stream+ argument must implement:
+<tt>read, write, flush, close, close_read, close_write, closed?</tt>
```

Comment:
How about `wait_readable` and `wait_writable`?

Reply:
Yes we can introduce those methods, but then we would also need to introduce `read_nonblock` and `write_nonblock` and all those semantics which don't make much sense for HTTP/2.

Reply:
That makes sense.  We shouldn't introduce them.